### PR TITLE
VIH-8571 Incorrect mute status using cache

### PR DESCRIPTION
**Pull request by PrasannaDommalapati**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/VIH-8571

### Change description ###
-used cahe

**Does this PR introduce a breaking change?** (check one with "x")

```
[x] Yes
[ ] No
```

### CODE REVIEW

File: VideoWeb/VideoWeb.Common/Models/ConferenceVideoControlStatuses.cs [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+        public Dictionary<string, VideoControlStatus> ParticipantIdToVideoControlStatusMap { get; set; }
+        
+        public int CompareTo(ConferenceVideoControlStatuses comparisonConferenceVideoControlStatuses)
+        {
+            bool areEqual = true;
+            foreach (var (key, videoControlStatus) in ParticipantIdToVideoControlStatusMap)
+            {
+                var comparisonvideoControlStatus = comparisonConferenceVideoControlStatuses.ParticipantIdToVideoControlStatusMap[key];
+                areEqual = areEqual && videoControlStatus.CompareTo(comparisonvideoControlStatus) > 0;
+            }
```
Comment (FarronF) [inline VideoWeb/VideoWeb.Common/Models/ConferenceVideoControlStatuses.cs:22]: This could be changed to stop comparing once one that is not equal is found.

Comment (omio-ma) [reply VideoWeb/VideoWeb.Common/Models/ConferenceVideoControlStatuses.cs:22]: so do you mean to return after areEqual is false?

**Review by ZJianWang — COMMENTED** [review]
could you please remove the UpdateConferenceCache in IParticipantService and any test for it, I believe no place using this anymore

File: VideoWeb/VideoWeb/Services/IParticipantService.cs [hunk 2]
```diff
@@ -79,5 +84,23 @@ public async Task UpdateConferenceCache(ConferenceDetailsResponse response)
             await _conferenceCache.UpdateConferenceAsync(conference);
```
Comment (ZJianWang) [inline VideoWeb/VideoWeb/Services/IParticipantService.cs:84]: I mean this method should be removed

File: VideoWeb/VideoWeb.Common/Models/ConferenceVideoControlStatuses.cs (hunk 1 above)
Comment (FarronF) [reply VideoWeb/VideoWeb.Common/Models/ConferenceVideoControlStatuses.cs:22]: Yeah that's what I was thinking. If the current .CompareTo() is 0 then return 0. And return 1 after the loop. 

This isn't a showstopper or anything.

File: VideoWeb/VideoWeb/ClientApp/src/app/services/conference/distributed-video-control-cache.service.ts [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+        if (!hearingControlStates) {
+            return null;
+        }
+
+        const setConferenceVideoControlStatusesRequest = {
+            participant_id_to_video_control_status_map: {}
+        } as SetConferenceVideoControlStatusesRequest;
+
+        if (hearingControlStates) {
+            for (const key in hearingControlStates.participantStates) {
```
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/services/conference/distributed-video-control-cache.service.ts:35]: Is key here participant id? If so might be good to give it a more descriptive name.

File: VideoWeb/VideoWeb.Common/Models/ConferenceVideoControlStatuses.cs (hunk 1 above)
Comment (omio-ma) [reply VideoWeb/VideoWeb.Common/Models/ConferenceVideoControlStatuses.cs]: I think it makes sense, made the code change requested

File: VideoWeb/VideoWeb/ClientApp/src/app/services/conference/video-control-cache-local-storage.service.ts [hunk 4]
```diff
@@ -29,7 +29,7 @@ export class VideoControlCacheLocalStorageService implements IVideoControlCacheS
         state[conferenceId] = hearingControlStates;
 
         this.localStorageService.save(this.localStorageKey, state);
-        return true;
+        return of(void 0);
```
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/services/conference/video-control-cache-local-storage.service.ts:32]: What's the purpose of this change? Could the method not return anything instead? Or could we set the return type as an observable?

File: VideoWeb/VideoWeb/ClientApp/src/app/services/conference/distributed-video-control-cache.service.ts (hunk 3 above)
Comment (PrasannaDommalapati) [reply VideoWeb/VideoWeb/ClientApp/src/app/services/conference/distributed-video-control-cache.service.ts:35]: @FarronF its a participant Id

File: VideoWeb/VideoWeb/ClientApp/src/app/services/conference/video-control-cache.service.ts [hunk 5]
```diff
@@ -29,11 +30,30 @@ export class VideoControlCacheService {
...
                     this.hearingControlStates = state;
+                    this.logger.info(`${this.loggerPrefix} initialised state for ${conference.id}.`, {
+                        hearingControlStates: this.hearingControlStates
+                    });
                 });
         });
     }
 
-    setSpotlightStatus(participantId: string, spotlightValue: boolean) {
+    refreshCache(): Observable<void> {
```
Comment (FarronF) [inline VideoWeb/VideoWeb/ClientApp/src/app/services/conference/video-control-cache.service.ts:40]: Is this used anywhere?

Comment (PrasannaDommalapati) [reply VideoWeb/VideoWeb/ClientApp/src/app/services/conference/video-control-cache.service.ts:40]: i don't think so it is used

File: VideoWeb/VideoWeb/ClientApp/src/app/services/conference/video-control-cache-local-storage.service.ts (hunk 4 above)
Comment (PrasannaDommalapati) [reply VideoWeb/VideoWeb/ClientApp/src/app/services/conference/video-control-cache-local-storage.service.ts:32]: this returns observable void

**Review by ZJianWang — APPROVED** [review]

**Review by omio-ma — APPROVED** [review]
